### PR TITLE
Fix https://github.com/pytorch/pytorch/issues/114892

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115054

### CODE REVIEW

File: torch/_functorch/aot_autograd.py [hunk 1]
```diff
@@ -451,6 +451,9 @@ def convert(idx, x):
                 if shape_env is not None:
                     from torch._dynamo.source import ConstantSource
                     if isinstance(x, int):
+                        # We always specialize on scalar values in export.
+                        if aot_config.is_export:
```
Comment (bdhirsh) [inline torch/_functorch/aot_autograd.py:455]: Does export want to do this unconditionally? Or do we (or in the future want to) provide a way for export users to mark scalar inputs as dynamic

**Review by bdhirsh — APPROVED** [review]

File: torch/_functorch/aot_autograd.py (hunk 1 above)
Comment (tugsbayasgalan) [reply torch/_functorch/aot_autograd.py:455]: We want to do it eventually (maybe next half?)

**Review by thiagocrepaldi — COMMENTED** [review]
Thank you! It would be great if we had an issue to track the "provide a way for export users to mark scalar inputs as dynamic" :)

**Pull request closed without merging**